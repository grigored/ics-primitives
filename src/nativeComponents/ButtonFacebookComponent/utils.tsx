// /**
//  * Created by alexbuicescu on 21 - Mar 2018.
//  */
//
// import { AllProps } from './ButtonFacebookComponent.types';
//
// declare const FB: any;
//
// export function fbComponentDidMount( props: AllProps ) {
//     let {fbLoadSdk, fbSdkLoaded, fbStatusChangeCallback} = props;
//     // let fbAppId = params && params.facebook_app_id!;
//     let fbAppId = '123';
//     // let inviteCode = this.props.location.query.code;
//     fbLoadSdk(fbAppId, fbSdkLoaded, fbStatusChangeCallback, props.referralCode);
// }
//
// export function fbOnClick( props: AllProps ) {
//     FB.login(null, {scope: 'email, user_friends'});
// }
//
// export function fbSignOut() {
//     if (FB && FB.getAuthResponse()) {
//         FB.logout(( response: any ) => {
//         });
//     }
// }