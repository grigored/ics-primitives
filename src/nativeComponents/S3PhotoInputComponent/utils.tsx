/**
 * Created by alexbuicescu on 18 - Aug 2017.
 */

import { FieldStateProps } from '../../redux/FormComponents/FormComponents.types';
import {checkImageRestrictions} from "./common";
import { ConnectedProps, S3PhotoInputComponentProps } from './S3PhotoInputComponent.types';
import { S3PhotoComponentDBValue } from './S3PhotoInputComponent.types';


declare global {
    interface Window {
        webkitURL?: string;
    }
}

export function onDrop(files: Array<File>, props: S3PhotoInputComponentProps & ConnectedProps & FieldStateProps<S3PhotoComponentDBValue>) {
    let {
        field, lastPhotoIndex, multiple, postPhotoToS3, showAlert,
        minWidth, maxWidth, minHeight, maxHeight, fixedWidth, fixedHeight, ratioWidthHeight,
    } = props;
    let file = files[0];
    let _URL = window.URL || window.webkitURL;

    if (!file.type.match('image.*')) {
        showAlert('FILE_IS_NOT_IMAGE');
        return;
    }

    let img = new Image();
    img.onload = function () {
        if (checkImageRestrictions(
            img.width, img.height, minWidth, maxWidth, minHeight, maxHeight, fixedWidth, fixedHeight, ratioWidthHeight
            )) {
            // do something

            let fieldData = {
                field: multiple ? `${field},${lastPhotoIndex}.url` : field,
                s3ExtraData: getPhotoDetails({rawValue: file}),
                preview: getImagePreview({rawValue: file})
            };

            postPhotoToS3(fieldData);
        }
    };
    img.onerror = function () {
        //setErrorForField(formName, field, _t(FILE_IS_NOT_IMAGE));
        showAlert('FILE_IS_NOT_IMAGE');
    };
    img.src = _URL.createObjectURL(file);
}

export function getPhotoDetails(formData: {rawValue: {type: string}}) {
    return {
        file: formData.rawValue,
        'Content-Type': formData.rawValue.type,
    }
}

export function getImagePreview(formData: {rawValue: any}): string {
    if (formData && formData.rawValue) {
        return formData.rawValue.preview || formData.rawValue.url || formData.rawValue;
    }
    return "";
}
