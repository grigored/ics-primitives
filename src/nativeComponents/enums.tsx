export enum BUTTON_TYPE {
    RAISED,
    FLAT,
}