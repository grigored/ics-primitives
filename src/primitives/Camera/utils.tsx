
export function checkForCameraPermission(permissionType: any, showAlert: Function): Promise<boolean> {
    // not used on web
    // return new Promise<boolean>(() => {});
    return new Promise<boolean>(() => {});
}

export function goToSettings(props: any) {

}
