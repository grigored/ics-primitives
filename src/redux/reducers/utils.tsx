
export function getUserDataLocalStorageName(): string {
    return 'userData';
}
