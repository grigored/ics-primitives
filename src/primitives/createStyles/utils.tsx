import * as React from "react";
import { StyleRules, WithStyles, StyleSheetClasses } from "../../utils/theme.types";
import { hoistNonReactStatics } from "../../lib/hoist-non-react-statics";
import {combineStyles } from "../../utils/combineStyles";


export function createStylesGeneric<T>(
    styles: StyleRules | (() => StyleRules),
    componentName: string,
    WrappedComponent: React.ComponentType<T & WithStyles>,
    StyleSheetCreate: (styles: StyleRules) => StyleSheetClasses,
): React.ComponentType<T> {

    class Enhance extends React.Component<T & {classes: any}> {

        render() {
            return (
                <WrappedComponent
                    classes={StyleSheetCreate(combineStyles(styles, componentName))}
                    {...this.props}
                />
            );
        }
    }

    // need this for statics like react-native-navigation navigatorStyle/ navigatorButtons
    hoistNonReactStatics(Enhance, WrappedComponent);
    return Enhance;

}
