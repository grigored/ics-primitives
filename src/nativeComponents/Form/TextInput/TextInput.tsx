import * as React from 'react';
import {FormControl, FormHelperText} from 'material-ui/Form';
import TextField from 'material-ui/TextField';
import { TextInputProps } from "./TextInput.types";
import { TEXT_INPUT_TYPES } from '../../../utils/enums';

const getKeyboardType = (inputType?: TEXT_INPUT_TYPES): string => {
    switch (inputType) {
        case TEXT_INPUT_TYPES.PASSWORD:
            return 'password';
        case TEXT_INPUT_TYPES.EMAIL:
            return 'email';
        case TEXT_INPUT_TYPES.PHONE:
            return 'tel';
        default:
            return 'text';
    }
};

export const TextInput = ({
    placeholder,
    inputType,
    onBlur,
    title,
    error,
    id,
    multiline,
    input,
}: TextInputProps) => (
    <FormControl fullWidth>
        <TextField
            id={id}
            value={input.value || ''}
            error={!!error}
            multiline={multiline}
            placeholder={placeholder || ''}
            label={title || ''}
            type={getKeyboardType(inputType)}
            onChange={(ev: React.ChangeEvent<HTMLInputElement>) =>
                !!input.onChange && input.onChange(ev.target.value)
            }
            onBlur={() => {
                onBlur && onBlur();
            }}
            helperText={error}
        />
        {false && <FormHelperText>some helping text here</FormHelperText>}
    </FormControl>
);