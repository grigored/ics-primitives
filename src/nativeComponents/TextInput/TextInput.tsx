import { FormHelperText } from "material-ui";
import { FormControl } from 'material-ui/Form';
import TextField from 'material-ui/TextField';
import * as React from 'react';
import {
    defaultDbToRaw, defaultRawToDb, getError,
    getKeyboardType
} from "src/nativeComponents/TextInput/textInputUtils";
import { createStyles } from '../../primitives/createStyles/createStyles';
import { FieldStateProps } from '../../redux/FormComponents/FormComponents.types';
import { TEXT_INPUT_TYPES } from "../../utils/enums";
import { appTheme } from '../../utils/theme';
import { WithStyles } from '../../utils/theme.types';
import { TextInputDBValue, TextInputProps } from "./TextInput.types";

export const INVALID_JSON_STRING = 'Invalid JSON string';
export const FIELD_MUST_BE_NUMBER = 'Field must be a number';
const styles = () => ( {
    underline: {
        '&:after': {
            backgroundColor: appTheme.textInputUnderlineColor,
        },
        '&:before': {
            backgroundColor: appTheme.textInputUnderlineColor,
        },
        '&:hover:not($disabled):before': {
            backgroundColor: `${appTheme.textInputUnderlineColor} !important`,
        },
        '&:hover:not($disabled):after': {
            backgroundColor: `${appTheme.textInputUnderlineColor} !important`,
        },
    },
    underlineError: {},
    input: {
        color: appTheme.textColor,
    },
} );

export class CTextInput extends React.PureComponent<TextInputProps & WithStyles & FieldStateProps<TextInputDBValue>, { rawValue: string, }> {
    componentWillMount() {
        let { value, inputType = TEXT_INPUT_TYPES.TEXT, dbToRaw, } = this.props;
        if (value !== null && value !== undefined) {
            this.setState( {
                rawValue: !!dbToRaw
                    ? dbToRaw( value )
                    : defaultDbToRaw( inputType, value )
            } )
        } else {
            this.setState( { rawValue: '' } )
        }
    }

    render() {
        let {
            onChange, placeholder, inputType, onBlur, title, error, id, multiline,
            disableUnderline, classes, rawToDb,
        } = this.props;
        return (
            <FormControl fullWidth>
                <TextField
                    id={id}
                    value={this.state.rawValue || ''}
                    error={!!error}
                    multiline={multiline}
                    placeholder={placeholder || ''}
                    label={title || ''}
                    type={getKeyboardType( inputType )}
                    onChange={( ev: React.ChangeEvent<HTMLInputElement> ) => {
                        let rawValue = ev.target.value;
                        let dbValue = !!rawToDb
                            ? rawToDb( rawValue )
                            : defaultRawToDb( inputType, rawValue );
                        this.setState( { rawValue: rawValue } );
                        let fieldError = getError( inputType, rawValue );
                        !!onChange && onChange(
                            !!fieldError
                                ? { value: dbValue, error: fieldError }
                                : dbValue
                        )
                    }}
                    onBlur={() => {
                        onBlur && onBlur();
                    }}
                    helperText={error}
                    InputProps={{
                        disableUnderline,
                        classes: {
                            input: classes.input as any,
                            underline: !!error ? classes.underlineError : classes.underline as any,
                        },
                    }}
                    InputLabelProps={{
                        // shrink: true,
                        style: {
                            color: appTheme.textInputLabelColor,
                        },
                    }}
                />
                {false && <FormHelperText>some helping text here</FormHelperText>}
            </FormControl>
        );
    }
}

export const TextInput: React.ComponentType<TextInputProps & FieldStateProps<TextInputDBValue>> = createStyles(
    styles,
    'TextInput',
    CTextInput
);
