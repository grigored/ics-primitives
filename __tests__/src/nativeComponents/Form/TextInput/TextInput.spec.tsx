import * as Enzyme from "enzyme";
import { mount } from "enzyme";
import * as Adapter from 'enzyme-adapter-react-16';
import * as React from 'react'
import { FIELD_MUST_BE_NUMBER } from "../../../../../src/utils/strings";
import { TextInput } from "src/nativeComponents/TextInput/TextInput";
import { TEXT_INPUT_TYPES } from "src/utils/enums";

Enzyme.configure( { adapter: new Adapter() } );

describe( '<TextInput />', () => {
    it( 'renders', () => {
        const wrapper = mount(
            <TextInput
                onChange={() => {}}
                inputType={TEXT_INPUT_TYPES.TEXT}
            />
        );
        expect( wrapper ).toMatchSnapshot();
    } );

    it( 'returns entered string', () => {
        let x = '';
        const wrapper = mount(
            <TextInput
                onChange={( value: any ) => x = value}
                inputType={TEXT_INPUT_TYPES.TEXT}
            />
        );

        const input: any = wrapper.find( 'input' );
        input.instance().value = 'First';
        input.simulate( 'change', input );
        expect( x ).toBe( 'First' );
    } );

    it( 'properly checks floats', () => {

        let x = '';
        const wrapper = mount(
            <TextInput
                onChange={( value: any ) => x = value}
                inputType={TEXT_INPUT_TYPES.FLOAT}
            />
        );
        const input: any = wrapper.find( 'input' );

        input.instance().value = '123';
        input.simulate( 'change', input );
        expect( x ).toBe( 123 );

        input.instance().value = '123.5';
        input.simulate( 'change', input );
        expect( x ).toBe( 123.5 );

        input.instance().value = '123.5.';
        input.simulate( 'change', input );
        expect( x['error'] ).toBe( FIELD_MUST_BE_NUMBER );
    } );

    it( 'initialises from value & uses rawToDb and dbToRaw the way it should', () => {
        let x = '';
        const wrapper = mount(
            <TextInput
                value={"123"}
                rawToDb={value => +value + 1}
                dbToRaw={value => ( value - 1 ).toString()}
                onChange={( value: any ) => x = value}
                inputType={TEXT_INPUT_TYPES.FLOAT}
            />
        );
        const input: any = wrapper.find( 'input' );
        expect( input.instance().value ).toBe( "122" );

        input.instance().value = '213';
        input.simulate( 'change', input );
        expect( x ).toBe( 214 );
    } );
} );