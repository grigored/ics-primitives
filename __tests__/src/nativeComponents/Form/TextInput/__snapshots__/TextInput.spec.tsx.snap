// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> renders 1`] = `
<CTextInput
  inputType={0}
  onChange={[Function]}
>
  <WithStyles(FormControl)
    fullWidth={true}
  >
    <FormControl
      classes={
        Object {
          "fullWidth": "MuiFormControl-fullWidth-4",
          "marginDense": "MuiFormControl-marginDense-3",
          "marginNormal": "MuiFormControl-marginNormal-2",
          "root": "MuiFormControl-root-1",
        }
      }
      component="div"
      disabled={false}
      error={false}
      fullWidth={true}
      margin="none"
      required={false}
    >
      <div
        className="MuiFormControl-root-1 MuiFormControl-fullWidth-4"
        onBlur={[Function]}
        onFocus={[Function]}
        required={false}
      >
        <TextField
          error={false}
          label=""
          onBlur={[Function]}
          onChange={[Function]}
          placeholder=""
          required={false}
          select={false}
          type="text"
          value=""
        >
          <WithStyles(FormControl)
            error={false}
            onBlur={[Function]}
            required={false}
          >
            <FormControl
              classes={
                Object {
                  "fullWidth": "MuiFormControl-fullWidth-4",
                  "marginDense": "MuiFormControl-marginDense-3",
                  "marginNormal": "MuiFormControl-marginNormal-2",
                  "root": "MuiFormControl-root-1",
                }
              }
              component="div"
              disabled={false}
              error={false}
              fullWidth={false}
              margin="none"
              onBlur={[Function]}
              required={false}
            >
              <div
                className="MuiFormControl-root-1"
                onBlur={[Function]}
                onFocus={[Function]}
                required={false}
              >
                <WithStyles(Input)
                  onChange={[Function]}
                  placeholder=""
                  type="text"
                  value=""
                >
                  <Input
                    classes={
                      Object {
                        "disabled": "MuiInput-disabled-10",
                        "error": "MuiInput-error-8",
                        "focused": "MuiInput-focused-9",
                        "formControl": "MuiInput-formControl-6",
                        "fullWidth": "MuiInput-fullWidth-13",
                        "inkbar": "MuiInput-inkbar-7",
                        "input": "MuiInput-input-14",
                        "inputDense": "MuiInput-inputDense-15",
                        "inputDisabled": "MuiInput-inputDisabled-16",
                        "inputMultiline": "MuiInput-inputMultiline-18",
                        "inputSearch": "MuiInput-inputSearch-19",
                        "inputSingleline": "MuiInput-inputSingleline-17",
                        "multiline": "MuiInput-multiline-12",
                        "root": "MuiInput-root-5",
                        "underline": "MuiInput-underline-11",
                      }
                    }
                    disableUnderline={false}
                    fullWidth={false}
                    multiline={false}
                    onChange={[Function]}
                    placeholder=""
                    type="text"
                    value=""
                  >
                    <div
                      className="MuiInput-root-5 MuiInput-formControl-6 MuiInput-inkbar-7 MuiInput-underline-11"
                      onBlur={[Function]}
                      onFocus={[Function]}
                    >
                      <input
                        aria-invalid={false}
                        aria-required={false}
                        className="MuiInput-input-14 MuiInput-inputSingleline-17"
                        disabled={false}
                        onChange={[Function]}
                        placeholder=""
                        type="text"
                        value=""
                      />
                    </div>
                  </Input>
                </WithStyles(Input)>
              </div>
            </FormControl>
          </WithStyles(FormControl)>
        </TextField>
      </div>
    </FormControl>
  </WithStyles(FormControl)>
</CTextInput>
`;
